addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,401 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.robonect.handler;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.robonect.RobonectClient;
+import org.openhab.binding.robonect.RobonectCommunicationException;
+import org.openhab.binding.robonect.RobonectEndpoint;
+import org.openhab.binding.robonect.config.RobonectConfig;
+import org.openhab.binding.robonect.model.ErrorEntry;
+import org.openhab.binding.robonect.model.MowerInfo;
+import org.openhab.binding.robonect.model.MowerStatus;
+import org.openhab.binding.robonect.model.Name;
+import org.openhab.binding.robonect.model.RobonectAnswer;
+import org.openhab.binding.robonect.model.VersionInfo;
+import org.openhab.binding.robonect.model.cmd.ModeCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.robonect.RobonectBindingConstants.CHANNEL_ERROR_CODE;
```

Comment:
I use IntelliJ ):-)  ... So probably it will do when I finally will reformat everything using eclipse.